src/main/menu.js

```diff
@@ -0,0 +1,89 @@
+import fs from 'fs'
+import path from 'path'
+import { app, Menu } from 'electron'
+import configureMenu, { dockMenu } from './menus'
+import { isFile, ensureDir, getPath, log } from './utils'
+
+const MAX_RECENTLY_USED_DOCUMENTS = 12
+const FILE_NAME = 'recently-used-documents.json'
+const recentlyUsedDocumentsPath = path.join(getPath('userData'), FILE_NAME)
+const isOsxOrWindows = /darwin|win32/.test(process.platform)
+
+export const addRecentlyUsedDocuments = filePath => {
+  if (isOsxOrWindows) app.addRecentDocument(filePath)
+  if (process.platform === 'darwin') return
+
+  let recentDocuments = getRecentlyUsedDocuments()
+  const index = recentDocuments.indexOf(filePath)
+  let needSave = index !== 0
```

Comment:
I am a little confused.

>if (index !== 0)

is it meaning that filePath is not in `recentDocuments` or this index great that `0`.

I think if `index` equal `-1`, then we need to save it to  `recently-used-documents.json` ?

Reply:
It means the file is not the last recently used file. The file does not exist in the list or it exists, but is not the last recently used file.

Reply:
Ok, I need think again. and you are right!